sdk/core/core-amqp/src/amqpAnnotatedMessage.ts

```diff
@@ -76,6 +95,21 @@ export const AmqpAnnotatedMessage = {
       application_properties: msg.applicationProperties,
       footer: msg.footer,
     };
-    return message;
+
+    // There is a loss of fidelity in the TTL header if larger than uint32 max value. As a workaround
+    // we set the absolute_expiry_time and creation_time on the message based on the TTL. These
+    // values are then used to reconstruct the accurate TTL for received messages.
+    if (msg.header?.timeToLive) {
+      const ttl = msg.header.timeToLive;
+      rhMsg.ttl = ttl > Constants.maxUint32Value ? Constants.maxUint32Value : ttl;
```

Comment:
nitpick: could also apply in other places:
```suggestion
      rhMsg.ttl = Math.min(Constants.maxUint32Value, ttl);
```